### DISCUSSION TITLE
Prometheus TSDB snapshots from multiple systems to single prometheus server

### DISCUSSION THREAD

**Discussion by rajdevworks** [prometheus/prometheus#10940] [Q&A]
Hello,

I have N systems in my cluster and collect a prometheus TSDB snapshot as required. The cluster is in isolated environment but the snapshot can be extracted to another environment where prometheus/grafana are running.

I'm able to visualize single snapshot by placing the data under /data directory and running prometheus.
If I have multiple TSDB snapshots from different systems, how can it be stored on Prometheus server such that they can be queried individually?
How can I apply external labels in Prometheus snapshot which can be translated to Grafana variables to filter on the Grafana dashboard?

**Comment by amver3108** [comment]
Can you pleaze share how to take backup from prometheus server directly ?
